### PR TITLE
Remove cmake option to enable examples in release candidate 3.0

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Comment by ndellingwood** [comment]
Cross-referencing #2688 
This PR removes out-dated examples and cmake option to build said example.

**Comment by masterleinad** [comment]
I would be happier if this pull request only removes the option to build the examples from the build system and we cherry-pick the actual changes to the `examples` from #2688 (once that is merged) to keep `develop` and `release-candidate-3.0` in sync.

**Comment by ndellingwood** [comment]
Slack-chatted earlier with @crtrott, merging this in to relaunch testing

File: example/CMakeLists.txt [hunk 1]
```diff
@@ -2,19 +2,19 @@
...
+#KOKKOS_ADD_EXAMPLE_DIRECTORIES(fenl)
+#KOKKOS_ADD_EXAMPLE_DIRECTORIES(multi_fem)
+#KOKKOS_ADD_EXAMPLE_DIRECTORIES(md_skeleton)
+#KOKKOS_ADD_EXAMPLE_DIRECTORIES(global_2_local_ids)
+#KOKKOS_ADD_EXAMPLE_DIRECTORIES(grow_array)
+#KOKKOS_ADD_EXAMPLE_DIRECTORIES(sort_array)
+#if(NOT Kokkos_ENABLE_CUDA)
+#KOKKOS_ADD_EXAMPLE_DIRECTORIES(tutorial)
+#endif()
+#KOKKOS_SUBPACKAGE_POSTPROCESS()
```
Comment (masterleinad) [inline example/CMakeLists.txt]: I am still not quite sure I understand why just commenting (especially the examples already deleted in this pull request) is the best approach. Why not just delete the file and all the other `CMakeLists.txt` files altogether?

**Pull request merged**